test/shared_test/lib_battery_voltage_test.cpp

```diff
@@ -37,16 +37,67 @@ TEST_F(voltage_dynamic_lib_battery_voltage_test, updateCapacityTest){
     cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 3
     model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
     EXPECT_NEAR(model->cell_voltage(), 3.9, tol);
+    EXPECT_NEAR(cap->q0(), 3, tol);
 
     I = -2;
     cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 5
     model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
     EXPECT_NEAR(model->cell_voltage(), 4.1, tol);
+    EXPECT_NEAR(cap->q0(), 5, tol);
 
     I = 5;
     cap->updateCapacity(I, dt_hour); // qmx = 10, I = 4.5, q0 = 0.5
     model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
     EXPECT_NEAR(model->cell_voltage(), 2.49, tol);
+    EXPECT_NEAR(cap->q0(), 0.5, tol);
+
+}
+
+TEST_F(voltage_dynamic_lib_battery_voltage_test, updateCapacitySubHourly){
+    double dt_hour = 1. / 2;
+    CreateModel(dt_hour);
+
+    double I = 2;
+    cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 3
+    model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
+    EXPECT_NEAR(model->cell_voltage(), 3.98, tol);
+    EXPECT_NEAR(cap->q0(), 4, tol);
+
+    I = -2;
+    cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 5
+    model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
+    EXPECT_NEAR(model->cell_voltage(), 4.1, tol);
+    EXPECT_NEAR(cap->q0(), 5, tol);
+
+    I = 5;
+    cap->updateCapacity(I, dt_hour); // qmx = 10, I = 4.5, q0 = 0.5
+    model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
+    EXPECT_NEAR(model->cell_voltage(), 3.78, tol);
+    EXPECT_NEAR(cap->q0(), 2.5, tol);
+
+}
+
+TEST_F(voltage_dynamic_lib_battery_voltage_test, updateCapacitySubMinute){
+    double dt_hour = 1. / 200;
+    CreateModel(dt_hour);
+
+    double I = 2;
+    cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 3
+    model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
+    EXPECT_NEAR(model->cell_voltage(), 4.01, tol);
+    EXPECT_NEAR(cap->q0(), 5, tol);
+
+    I = -2;
+    cap->updateCapacity(I, dt_hour); // qmx = 10, q0 = 5
+    model->updateVoltage(cap->q0(), cap->qmax(), cap->I(), 0, dt_hour);
+    EXPECT_NEAR(model->cell_voltage(), 4.1, tol);
+    EXPECT_NEAR(cap->q0(), 5, tol);
```

Comment:
Can this test have a smaller tolerance? It would be nice to see the small changes in q0.

Reply:
done